Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -373,7 +373,7 @@ open class ClientsSynchronizer: TimestampedSingleCollectionSynchronizer, Synchro
             >>== { self.processCommandsFromRecord(ours, withServer: storageClient) }
             >>== { (shouldUpload, commands) in
                 let isFirstSync = self.lastFetched == 0
-                return self.maybeUploadOurRecord(shouldUpload || self.why == .didLogin, ifUnmodifiedSince: ours?.modified, toServer: storageClient)
+                return self.maybeUploadOurRecord(shouldUpload || self.why == .didLogin || self.why == .clientNameChanged, ifUnmodifiedSince: ours?.modified, toServer: storageClient)
```

Comment:
I have a gentle preference for breaking this out into a descriptive variable:

```
let ourRecordDidChange = self.why == .didLogin || self.why == .clientNameChanged
…
```

and even putting a `why.requiresUpload()` method on the reason enum to encapsulate this.